Small changes

pgpleroux opened the pull request:
Fixed to work for:

* REHL8
* Puppet7
* ZendServer 2021.1.1

Changes:

* zendserver url has changed from 'http' to 'https'
* support for RHEL8 and PHP 7.4
* test for $join_cluster to suppress warning message

pgpleroux committed:
Fixed to work for:

* REHL8
* Puppet7
* ZendServer 2021.1.1

Changes:

* zendserver url has changed from 'http' to 'https'
* support for RHEL8 and PHP 7.4
* test for $join_cluster to suppress warning message

weierophinney commented on manifests/init.pp:
```diff
@@ -96,7 +96,7 @@
   validate_bool($manage_repos)
   validate_re($webserver, ['\Aapache|nginx\Z',], 'Only apache or nginx are supported.')
-  validate_re($phpversion, ['\A5.3|5.4|5.5|5.6|7.0|7.1\Z',], 'Only versions 5.4 to 7.1 are supported.')
+  validate_re($phpversion, ['\A5.3|5.4|5.5|5.6|7.0|7.1|7.2|7.3|7.4\Z',], 'Only versions 5.4 to 7.4 are supported.')
```
```suggestion
  validate_re($phpversion, ['\A5.6|7.1|7.2|7.3|7.4\Z',], 'Only versions 5.6 and 7.1 - 7.4 are supported.')
```

These are the only PHP versions supported by current and LTS versions of Zend Server.

weierophinney reviewed (changes requested):
One minor change, but otherwise :+1:

pgpleroux committed:
PHP: only versions 5.6 and 7.1 - 7.4 are supported.
Modified init.pp to use hiera lookups with defaults as per existing
code

pgpleroux committed:
Convert module to PDK compatible - eye candy only
Fixed PDK attempts to force a coding style on people :-)
Also fixed the stupid 140 character line width (reminds me of Fortran V)
But I know what will happen: at some stage Puppet is going to get buffer
overflows because someone is going to use 'char* buf[141]' instead of
'char* buf = (char*)malloc(1024)' in the parser (One of my favourite rants!)

pgpleroux committed:
Added README-hiera.md for using the module with hiera

ghost commented:
Done
